sim/firesim-lib/src/main/scala/endpoints/TracerVWidget.scala

```diff
@@ -97,7 +97,7 @@ class TracerVWidget(traceProto: Seq[Vec[DeclockedTracedInstruction]])(implicit p
 
   io.tReset.ready := tFireHelper.fire(io.tReset.valid)
   io.hPort.fromHost.hValid := tFireHelper.fire(io.hPort.fromHost.hReady)
-  io.hPort.toHost.hReady := tFireHelper.fire(io.hPort.toHost.hValid)
+  io.hPort.toHost.hReady := tFireHelper.fire
```

Comment:
Why do you not need to `hValid` anymore?

Reply:
This is actually related to the uncommented //Hack above. Here i've made ready depend on valid, since ready fans out to all of the token channels in that feed tokens into this endpoint. If one of those token channels has it's "sub-token" valid before the rest, it will be falsely dequeued. Really i should fuse the token properly by making the ready for each sub-token depend on AND {valids of all other sub-tokens} instead of doing it this way.